src/plugins/workspace/server/plugin.test.ts

```diff
@@ -205,31 +224,97 @@ describe('Workspace server plugin', () => {
       });
     });
 
-    it('without / request path', async () => {
+    it('with / request path and one workspace', async () => {
       const request = httpServerMock.createOpenSearchDashboardsRequest({
-        path: '/foo',
+        path: '/',
+      });
+      const workspaceSetup = await workspacePlugin.setup(setupMock);
+      const client = workspaceSetup.client;
+      jest.spyOn(client, 'list').mockResolvedValue({
+        success: true,
+        result: {
+          total: 1,
+          per_page: 100,
+          page: 1,
+          workspaces: [{ id: 'workspace-1', name: 'workspace-1' }],
+        },
       });
-      await workspacePlugin.setup(setupMock);
       const toolKitMock = httpServerMock.createToolkit();
 
       await registerOnPostAuthFn(request, response, toolKitMock);
-      expect(toolKitMock.next).toBeCalledTimes(1);
+      expect(response.redirected).toBeCalledWith({
+        headers: {
+          location: '/mock-server-basepath/w/workspace-1/app/workspace_navigation',
+        },
+      });
     });
 
-    it('with more than one workspace', async () => {
+    it('with / request path and more than one workspaces', async () => {
       const request = httpServerMock.createOpenSearchDashboardsRequest({
         path: '/',
       });
       const workspaceSetup = await workspacePlugin.setup(setupMock);
       const client = workspaceSetup.client;
       jest.spyOn(client, 'list').mockResolvedValue({
         success: true,
-        result: { total: 1 },
+        result: {
+          total: 2,
+          per_page: 100,
+          page: 1,
+          workspaces: [
+            { id: 'workspace-1', name: 'workspace-1' },
+            { id: 'workspace-2', name: 'workspace-2' },
+          ],
+        },
       });
       const toolKitMock = httpServerMock.createToolkit();
 
       await registerOnPostAuthFn(request, response, toolKitMock);
-      expect(toolKitMock.next).toBeCalledTimes(1);
+      expect(response.redirected).toBeCalledWith({
+        headers: {
+          location: '/mock-server-basepath/app/home',
+        },
+      });
+    });
+
+    it('with / request path and default workspace', async () => {
+      const request = httpServerMock.createOpenSearchDashboardsRequest({
+        path: '/',
+      });
+      const savedObjectsClient: jest.Mocked<SavedObjectsClientContract> = savedObjectsClientMock.create();
+      const mockUiSettings = {
+        get: jest.fn().mockResolvedValue('default'),
+      };
```

Comment:
For future-proofing, this should use `uiSettingsServiceMock` and not reinvent it.

Reply:
Sure, I have updated the code